collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -765,7 +765,15 @@ protected void onActivityResult(int requestCode, int resultCode, final Intent in
                 break;
 
             case RequestCodes.ARBITRARY_FILE_CHOOSER:
-                saveFileAnswer(intent.getData());
+
+                showDialog(SAVING_DIALOG);
+                Runnable saveFileRunnable = new Runnable() {
```

Comment:
It seems to me that you could let this drop down to the video case as you did with audio. I think the only difference is the name of the `Runnable`. `saveMediaRunnable` seems fine to me for all of them or you could go with the slightly more general `saveFileRunnable`.